src/frontend/components/Chat/SharedChatComponents/InputDisplayNameOverlay/index.tsx

```diff
@@ -0,0 +1,173 @@
+    id: 'components.InputBar.inputDisplayNamePlaceholder',
+  },
+  inputTooShortAlertMessage: {
+    defaultMessage: 'Min length is 3 characters.',
+    description:
+      'An alert message explaining why the entered display name is invalid.',
+    id: 'components.InputDisplayNameOverlay.inputTooShortAlertMessage',
+  },
+  inputTooLongAlertMessage: {
+    defaultMessage: 'Max length is 15 characters.',
+    description:
+      'An alert message explaining why the entered display name is invalid.',
+    id: 'components.InputDisplayNameOverlay.inputTooLongAlertMessage',
+  },
+});
+
+interface InputDisplayNameOverlayProps {
+  setInputBarActive: React.Dispatch<React.SetStateAction<boolean>>;
+  setOverlay: React.Dispatch<React.SetStateAction<boolean>>;
+}
+
+export const InputDisplayNameOverlay = ({
+  setInputBarActive,
+  setOverlay,
+}: InputDisplayNameOverlayProps) => {
+  const intl = useIntl();
+  const [alertsState, setAlertsState] = useState<string[]>([]);
+
+  const processDisplayName = (displayName: string) => {
+    const alerts: string[] = [];
+    if (displayName.toLowerCase().includes(ANONYMOUS_ID_PREFIX)) {
+      alerts.push(
+        intl.formatMessage(messages.inputAnonymousKeywordForbiddenAlertMessage),
+      );
+    }
+    if (displayName.length < 3) {
+      alerts.push(intl.formatMessage(messages.inputTooShortAlertMessage));
+    }
+    if (displayName.length > 15) {
+      alerts.push(intl.formatMessage(messages.inputTooLongAlertMessage));
+    }
+    if (alerts.length === 0) {
+      converse.claimNewNicknameInChatRoom(displayName);
+      setInputBarActive(true);
+      setOverlay(false);
+      return true;
+    } else {
+      setAlertsState(alerts);
+      return false;
+    }
+  };
+
+  const handleExitCrossClick = () => {
+    setOverlay(false);
+  };
+
+  return (
+    <Box height="100%">
+      <Box
+        direction="row-reverse"
+        margin={{
+          right: '5px',
+          top: '5px',
+        }}
+      >
+        <Box
+          onClick={handleExitCrossClick}
+          title={intl.formatMessage(messages.closeButtonTitle)}
+        >
+          <ExitCrossSVG
+            containerStyle={{
+              height: '20px',
+              width: '20px',
+            }}
+            iconColor={'#031963'}
+          />
+        </Box>
+      </Box>
+      <Box
+        margin={{
+          horizontal: '20px',
+          vertical: '30px',
+        }}
+        pad="3px"
+      >
+        <Box background="#edf5fa" gap="8px" pad="8px" round="6px">
+          <Box direction="row">
+            <Text
+              margin={{
+                right: '5px',
+              }}
+              size="14px"
+            >
+              {intl.formatMessage(messages.inputDisplayNameLabel)}
+            </Text>
+            <Box>
+              <Tip
+                content={
+                  <Box background="#ffffff" pad="2px" round="6px" width="150px">
+                    <Text size="10px">
```

Comment:
here aswell

Reply:
done in [fb4e6bc849131d7bd921ef7f0c208e52364de54a](https://github.com/openfun/marsha/pull/1276/commits/fb4e6bc849131d7bd921ef7f0c208e52364de54a)